src/abi_impl.rs

```diff
@@ -375,17 +372,17 @@ pub(crate) fn assembly_script_get_time(env: &Env) -> ABIResult<i64> {
 }
 
 /// Tooling, return a StringPtr allocated from a String
-fn pointer_from_string(env: &Env, value: &String) -> ABIResult<StringPtr> {
-    match StringPtr::alloc(value, &env.wasm_env) {
+fn pointer_from_string(env: &Env, value: &str) -> ABIResult<StringPtr> {
+    match StringPtr::alloc(&value.into(), &env.wasm_env) {
         Ok(ptr) => Ok(*ptr),
         Err(err) => abi_bail!(err),
     }
 }
 
 /// Tooling, return a StringPtr allocated from bytes with utf8 parsing
-fn pointer_from_utf8(env: &Env, value: &Vec<u8>) -> ABIResult<StringPtr> {
+fn pointer_from_utf8(env: &Env, value: &[u8]) -> ABIResult<StringPtr> {
     match std::str::from_utf8(value) {
-        Ok(data) => match StringPtr::alloc(&data.to_string(), &env.wasm_env) {
+        Ok(data) => match StringPtr::alloc(&data.into(), &env.wasm_env) {
```

Comment:
> If you need a String instead of a &str, consider String::from_utf8.

from https://doc.rust-lang.org/std/str/fn.from_utf8.html

Reply:
Ok, I just revert it finally :rofl: